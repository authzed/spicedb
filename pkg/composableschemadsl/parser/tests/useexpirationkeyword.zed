use expiration

caveat expiration(someparam int) {
    someparam == 42
}