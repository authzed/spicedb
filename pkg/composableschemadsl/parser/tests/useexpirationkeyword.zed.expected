NodeTypeFile
  end-rune = 21
  input-source = use expiration keyword test
  start-rune = 0
  child-node =>
    NodeTypeUseFlag
      end-rune = 13
      input-source = use expiration keyword test
      start-rune = 0
      use-flag-name = expiration
    NodeTypeCaveatDefinition
      end-rune = 21
      input-source = use expiration keyword test
      start-rune = 16
      child-node =>
        NodeTypeError
          end-rune = 21
          error-message = Expected identifier, found token TokenTypeKeyword
          error-source = expiration
          input-source = use expiration keyword test
          start-rune = 23
    NodeTypeError
      end-rune = 21
      error-message = Unexpected token at root level: TokenTypeKeyword
      error-source = expiration
      input-source = use expiration keyword test
      start-rune = 23